meter/tesla.go

```diff
@@ -101,10 +130,38 @@ func NewTesla(uri, usage string) (api.Meter, error) {
 	return decorateTesla(m, totalEnergy, batterySoC), nil
 }
 
+// Login calls login and saves the returned cookie
+func (m *Tesla) Login() error {
+	// username for the powerwall seems to always be customer; email is not required for authentication
+	payload := map[string]interface{}{"password": m.password, "username": "customer"}
+	payloadBytes, err := json.Marshal(payload)
+	if err != nil {
```

Comment:
if prefer using is err==nil for the next step but ymmv

Reply:
I prefer to not have much nested if statements. In my opinion it makes the code more readable to immediately return the error.